packages/react/src/auto/polaris/inputs/PolarisAutoEnumInput.tsx

```diff
@@ -0,0 +1,149 @@
+import type { ComboboxProps } from "@shopify/polaris";
+import { AutoSelection, Combobox, EmptySearchResult, Icon, InlineStack, Listbox, Tag, Text } from "@shopify/polaris";
+import { SearchIcon } from "@shopify/polaris-icons";
+import React, { useEffect, useMemo } from "react";
+import { type Control } from "react-hook-form";
+import { useEnumInputController } from "../../hooks/useEnumInputController.js";
+
+export const PolarisAutoEnumInput = (props: { field: string; control?: Control<any> } & Partial<ComboboxProps>) => {
+  const { field: fieldApiIdentifier, control, ...comboboxProps } = props;
+  const {
+    allowMultiple,
+    allowOther,
+    onSelectionChange,
+    selectedOptions,
+    allOptions,
+    filteredOptions,
+    searchQuery,
+    label,
+    metadata,
+    isError,
+    errorMessage,
+  } = useEnumInputController({ field: fieldApiIdentifier, control });
+  const { value: searchValue, setValue: setSearchValue } = searchQuery;
+
+  const selectedTagsElement = useMemo(
+    () =>
+      allowMultiple && selectedOptions.length > 0 ? (
+        <InlineStack gap="150">
+          {selectedOptions.map((tag) => (
+            <Tag key={`option-${tag}`} onRemove={() => onSelectionChange(tag)}>
+              {tag}
+            </Tag>
+          ))}
+        </InlineStack>
+      ) : null,
+    [allowMultiple, onSelectionChange, selectedOptions]
+  );
+
+  const optionItemElement = useMemo(() => {
+    if (allOptions.length === 0) return null;
+
+    const formatOptionText = (option: string) => {
+      const trimValue = searchValue.trim().toLocaleLowerCase();
+      const matchIndex = option.toLocaleLowerCase().indexOf(trimValue);
+
+      if (!searchValue || matchIndex === -1) return option;
+
+      const start = option.slice(0, matchIndex);
+      const highlight = option.slice(matchIndex, matchIndex + trimValue.length);
+      const end = option.slice(matchIndex + trimValue.length, option.length);
+
+      return (
+        <p>
+          {start}
+          <Text fontWeight="bold" as="span">
+            {highlight}
+          </Text>
+          {end}
+        </p>
+      );
+    };
+
+    return filteredOptions.map((option) => {
+      return (
+        <Listbox.Option key={option} value={option} selected={selectedOptions.includes(option)} accessibilityLabel={option}>
+          <Listbox.TextOption selected={selectedOptions.includes(option)}>{formatOptionText(option)}</Listbox.TextOption>
+        </Listbox.Option>
+      );
+    });
+  }, [allOptions.length, filteredOptions, searchValue, selectedOptions]);
+
+  const addExtraOptionElement = useMemo(() => {
+    if (!allowOther) return null;
+
+    const noResults = searchValue && !allOptions.includes(searchValue);
+    return noResults ? <Listbox.Action value={searchValue}>{`Add "${searchValue}"`}</Listbox.Action> : null;
```

Comment:
If `searchValue` is make up of only spaces, it will get compressed into a single space. Maybe replacing them all with a unicode-looking space that won't get lost will avoid this